cli/azd/pkg/pipeline/pipeline_manager.go

```diff
@@ -349,6 +374,10 @@ func (pm *PipelineManager) Configure(ctx context.Context) (result *PipelineConfi
 		additionalVariables[environment.ResourceGroupEnvVarName] = rgGroup
 	}
 
+	// Merge azd required variables and secrets with the ones defined on azure.yaml
+	additionalVariables, additionalSecrets = pm.configOptions.SecretsAndVars(
```

Comment:
Suppose I have `INPUT_FOO` defined in `os.env` but not saved in azd's environment. `INPUT_FOO` is required by `main.parameters.json` env substitution and will fail without it defined. What is the expected behavior here?

Reply:
I am hoping to start this new feature for `azd pipeline config` to be limited to the azd-environment.
We can easily pull from os.env as well (if the value is not found in azd .env), but I want to start with a mental-model where `azd pipeline config` takes your .env only to the pipeline.
I am trying to think about os.env as what it would be an `organization variable/secret` across all your repositories. And in the future, explore if azd could help you to set those cross-repository variables/secrets.

It is similar to how the remote-env is scoped to the .env file and it would just lost any os.env variables when moving from one device to another.